File: lingting-core/src/main/java/live/lingting/framework/money/MoneyConfigHolder.java
package live.lingting.framework.money;

import live.lingting.framework.thread.StackThreadLocal;
import lombok.experimental.UtilityClass;

/**
 * @author lingting 2023-05-07 18:00
 */
@UtilityClass
public class MoneyConfigHolder {

	private static final StackThreadLocal<MoneyConfig> THREAD_LOCAL = new StackThreadLocal<>();

	public static MoneyConfig get() {
		return THREAD_LOCAL.get();
	}

	public static void put(MoneyConfig config) {
		THREAD_LOCAL.put(config);
	}

	public static void pop() {
		THREAD_LOCAL.pop();
	}

}

File: lingting-core/src/main/kotlin/live/lingting/framework/money/MoneyConfigHolder.kt
package live.lingting.framework.money

import live.lingting.framework.context.StackContext

/**
 * @author lingting 2023-05-07 18:00
 */
object MoneyConfigHolder {

    private val CONTEXT = StackContext<MoneyConfig?>()

    @JvmStatic
    fun get(): MoneyConfig? {
        return CONTEXT.peek()
    }

    @JvmStatic
    fun put(config: MoneyConfig?) {
        CONTEXT.push(config)
    }

    @JvmStatic
    fun pop() {
        CONTEXT.pop()
    }

}
